controllers/validation/accountpool_validation_controller.go

```diff
@@ -0,0 +1,252 @@
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	cm, err := utils.GetOperatorConfigMap(r.Client)
+	if err != nil {
+		logs.Error(err, "Could not retrieve the operator configmap")
+		return utils.RequeueAfter(5 * time.Minute)
+	}
+
+	var accountPoolValidationEnabled bool = false
+
+	enabled, err := strconv.ParseBool(cm.Data["feature.accountpool_validation"])
+	if err != nil {
+		logs.Info("Could not retrieve feature flag 'feature.accountpool_validation' - accountpool validation is disabled")
+	} else {
+		accountPoolValidationEnabled = enabled
+	}
+	logs.Info("Is accountpool_validation enabled?", "enabled", accountPoolValidationEnabled)
+
+	reqLogger.Info("Checking ConfigMap for ServiceQuotas")
+	// check if accountpool has servicequota defined in configmap
+	reginalServiceQuotas, err := r.getAccountPoolRegionalServiceQuota(reqLogger, currentAccountPool.Name)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("Updating Account ServiceQuotas")
+	_, err = r.checkAccountServiceQuota(reqLogger, currentAccountPool.Name, reginalServiceQuotas, accountPoolValidationEnabled)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return utils.DoNotRequeue()
+}
+
+func (r *AccountPoolValidationReconciler) accountSpecUpdate(reqLogger logr.Logger, account *awsv1alpha1.Account) error {
+	err := r.Client.Update(context.TODO(), account)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Account spec update for %s failed", account.Name))
+	}
+	return err
+}
+
+func (r *AccountPoolValidationReconciler) accountStatusUpdate(reqLogger logr.Logger, account *awsv1alpha1.Account) error {
+	err := r.Client.Status().Update(context.TODO(), account)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Account status update for %s failed", account.Name))
+	}
+	return err
+}
+
+func (r *AccountPoolValidationReconciler) getAccountPoolAccounts(accountPoolName string) ([]awsv1alpha1.Account, error) {
+	//Get the number of actual unclaimed AWS accounts in the pool
+	accountList := &awsv1alpha1.AccountList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(awsv1alpha1.AccountCrNamespace),
+	}
+	if err := r.Client.List(context.TODO(), accountList, listOpts...); err != nil {
+		return nil, err
+	}
+	var accounts []awsv1alpha1.Account
+	for _, account := range accountList.Items {
+		// gets accounts belonging to poo
+		if account.IsOwnedByAccountPool() && account.Spec.AccountPool == accountPoolName {
+			accounts = append(accounts, account)
+		}
+	}
+	return accounts, nil
+
+}
+
+// Gets ServiceQuota from ConfigMap
+func (r *AccountPoolValidationReconciler) getAccountPoolRegionalServiceQuota(reqLogger logr.Logger, accountPoolName string) (awsv1alpha1.RegionalServiceQuotas, error) {
```

Comment:
This also seems to be mostly a copy of https://github.com/reedcort/aws-account-operator/blob/3e2296fe1b5c73a6a7ed93facda82cdb0c671242/controllers/accountpool/accountpool_controller.go#L117 - I'd rather not duplicate more code in AAO, the current code base is already hard to read.